docs/CHANGES_4.0.0.md

```diff
@@ -1,39 +1,249 @@
-# Changes in 4.x
+# Changes in 4.x (and how to migrate!)
 
-WIP
+_Hello dear reader, **thank you** for adopting version 4.x of the mongodb driver, from the bottom of our developer hearts we thank you so much for taking the time to upgrade to our latest and greatest offering of a stunning database experience.
+We hope you enjoy your upgrade experience and this guide gives you all the answers you are searching for.
+If anything, and we mean anything, hinders you upgrade experience please let us know via [JIRA](https://jira.mongodb.org/browse/NODE).
+We know breaking changes are hard but they are sometimes for the best.
+Anyway, enjoy the guide, see you at the end!_
 
-## Versioned API
+## Key Changes
 
-Versioned API is a new feature in MongoDB 5.0 that allows user-selectable API versions, subsets of MongoDB server semantics, to be declared on a client. During communication with a server, clients with a declared API version will force the server to behave in a manner compatible with the API version. Declaring an API version on a client can be used to ensure consistent responses from a server, providing long term API stability for an application. The declared API version is applied to all commands run through the client, including those sent through the generic RunCommand helper. Specifying versioned API options in the command document AND declaring an API version on the client is not supported and will lead to undefined behaviour.
+### Typescript
 
-### Declare an API version on a client:
+We've migrated the driver to Typescript! Users can now harness the power of type hinting and intellisense in editors that support it to develop their MongoDB applications. Users don't even have to adopt Typescript to get the benefits. Along with the type hinting there's consistent docs formatting that editors should be able to display while developing.
+Recently we migrated our BSON library to TypeScript as well, this version of the driver pulls in those changes.
+
+#### Community Types users (@types/mongodb)
+
+If you are a user of the community types (@types/mongodb) there will likely be compilation errors while adopting the types from our codebase.
+Unfortunately we could not achieve a one to one match in types due to the details of writing the codebase in Typescript vs definitions for the user layer API along with the breaking changes of this major version. Please let us know if there's anything that is a blocker to upgrading [on JIRA](https://jira.mongodb.org/browse/NODE).
+
+### NodeJS Version
+
+We now require node 12.9 or greater for version 4 of the driver.
+If that's outside your support matrix at this time, that's okay!
+Bug fix support for our 3.x branch will not be ending until summer 2022!
+
+### Cursor changes
+
+#### AbstractCursor, FindCursor, AggregationCursor, ChangeStreamCursor, ListCollectionsCursor
```

Comment:
This heading is confusing to me, what is it intended to convey?

Reply:
I guess just listing out the affected classes, I'll make it into a list instead